### ISSUE TITLE
[Fastlane.Swift] Runner hanging on failure 

### ISSUE THREAD

**Issue by jhandguy** [fastlane/fastlane#16677]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information

- Breaking version: `2.150.0.rc1`
- Last working version: `2.149.0`

### Regression Description

This behaviour happens for any lane (scan, etc..) which do not have an `error_callback`, and sometimes also the lanes that have an `error_callback` but only for certain kind of errors (i.e. `sh` with a command not found error).

Code Samples:
```
class Fastfile: LaneFile {
    func testLane() {
        sh(command: "echoo") { error in
            sh(command: "echo OK")
        }
    }
}
```

```
class Fastfile: LaneFile {
    func testLane() {
        sh(command: "ls stuff")
    }
}
```

**Comment by joshdholtz** [comment]
@minuscorp 👋 Hey! Do you mind taking a look at this whenever you get some time? 😊

**Comment by jhandguy** [comment]
There seems to be 2 cases where the new logic fails:
1. Any lane without an `error_callback` specified will hang if it fails
2. For lanes that have an `error_callback` specified, it will still hang for certain kind of errors (i.e. command not found)

My guess is that the logic for handling failure is the faulty part, and in the case of an `error_callback` specified, the server socket won't actually return an error but a readyForNext. That would explain why it works sometimes for lanes with an `error_callback`.

**Comment by minuscorp** [comment]
Yeah, the Swift part goes ahead with the execution if it receives a `ready_for_next` from the Ruby Socket, so the Ruby part for some reason is not getting the correct status code from the execution of each of this parts. As we left the Ruby part unchanged with the last PR, could you get a latter version of fastlane and test if this happens in the same way? (with no closures ofc). Until the weekend I'm quite out of time 😅

**Comment by minuscorp** [comment]
Thinking of it, I'm quite sure why is happening, but I'll need more time to fix it 🗡️

**Comment by jhandguy** [comment]
@minuscorp I was running it on the last version of Fastlane, I've put the breaking version as `2.149.0` because it was not happening before then, but my findings are based from `master` 👍 
Let me know if there's anything I can do to help apart from testing, I've already created a PR (https://github.com/fastlane/fastlane/pull/16675) for increasing the default socket timeout which we are reaching more easily now with the new App Store Connect API 😢 
It'd be great if you could take a look, I also did some refactoring/code styling! 👌

**Comment by minuscorp** [comment]
Let me first solve the issue, then we can jump into that!

**Comment by minuscorp** [comment]
Hey @jhandguy! I've been looking into the releases, and it seems that no public release has been done regarding Fastlane.swift but one with argument parsing from Ruby to Swift interface (which does not interfere in this issue). Can we confirm that my PR is not released yet @joshdholtz? 
Thank you both for the work!

**Comment by jhandguy** [comment]
Hi @minuscorp, yes your PR was released, I can confirm I was getting your changes when updating to 2.149.0 (that’s why I created this issue at the first place). I also saw your PR wasn’t added to the release notes though that’s weird 🤔

**Comment by minuscorp** [comment]
You saw another PR I worked with in that week 😅

**Comment by jhandguy** [comment]
Oh you're right @minuscorp, my bad, it was introduced in `2.150.0.rc1` not `2.149.1`, I'll update the description! 
None the less since the new App Store API and everybody needing to update to `2.150.rc3` now, I'd still count it as released (maybe unwillingly?) 🤔

**Comment by jhandguy** [comment]
Thanks, for the awesome work BTW, it's great to see Fastlane Swift moving forward! 💪

**Comment by jhandguy** [comment]
Fixed in `2.150.0`! 🎉
